### ISSUE TITLE
[Question] How to use Request to modify createOne mutation input

### ISSUE THREAD

**Issue by josheleonard** [doug-martin/nestjs-query#326]
I have a demo app with Tasks (Todos) that are owned by a user. 
is there a declarative way to replace a DTO field value with a variable attached to the Request (like the user id)?

A persist option is is used for this in nestjsx/crud's decorator: https://github.com/nestjsx/crud/wiki/Controllers#request-authentication

Is there a similar option in this library?

**Comment by mmChuckie** [comment]
What's the status on this issue? Got the same problem as I have an application that relies on a lot of user input made by authenticated users. :-)

**Comment by doug-martin** [comment]
I'll be releasing this in the next release.

I'm currently exploring an option to allow specifying different "hooks" that will run when the `create` or `update` hooks are run.

For example

```ts
@InputType('TodoItemInput')
@BeforeCreateOne((input: CreateOneInputType<TodoItemDTO>, context) => {
  input.input.createdBy = getUserName(context);
  return input;
})
@BeforeCreateMany((input: CreateManyInputType<TodoItemDTO>, context) => {
  const createdBy = getUserName(context);
  input.input = input.input.map((c) => ({ ...c, createdBy }));
  return input;
})
export class TodoItemInputDTO {
  @IsString()
  @MaxLength(20)
  @Field()
  title!: string;

  @IsBoolean()
  @Field()
  completed!: boolean;
}

```

This is still in POC stage but I hope to have a release by the end of the week.

**Comment by doug-martin** [comment]
@josheleonard @mmChuckie I just released `v0.17.4` which add the hooks described above.

You can check out the docs [here](https://doug-martin.github.io/nestjs-query/docs/graphql/hooks)
The [`typeorm` example](https://github.com/doug-martin/nestjs-query/tree/master/examples/typeorm/src) has been updated to use the new decorators also.

Thanks again!
